pkg/client/clientcmd/api/types.go

```diff
@@ -26,7 +25,11 @@ import (
 
 // Config holds the information needed to build connect to remote kubernetes clusters as a given user
 type Config struct {
-	api.TypeMeta `json:",inline"`
+	// Legacy field from pkg/api/types.go TypeMeta.
+	// TODO(jlowdermilk): remove this after eliminating downstream dependencies.
+	Kind string `json:"kind,omitempty"`
+	// Version of the schema for this config object.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
I don't understand - if this object is not versioned with the API, what
does APIVersion represent?  if it is versioned, it should live with the
versioned APIs, no?

On Fri, May 15, 2015 at 4:07 PM, Jeff Lowdermilk <notifications@github.com>
wrote:

> In pkg/client/clientcmd/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8344#discussion_r30452914>
> :
>
> > @@ -26,7 +25,11 @@ import (
> >
> >  // Config holds the information needed to build connect to remote kubernetes clusters as a given user
> >  type Config struct {
> > -	api.TypeMeta `json:",inline"`
> > +	// Legacy field from pkg/api/types.go TypeMeta.
> > +	// TODO(jlowdermilk): remove this after eliminating downstream dependencies.
> > +	Kind string `json:"kind,omitempty"`
> > +	// Version of the schema for this config object.
> > +	APIVersion string `json:"apiVersion,omitempty"`
>
> From a design point of view, I think APIVersion is more important to keep
> than kind. Clients that consume kubeconfig generally don't consume other
> objects. But the practical reason for keeping both is to avoid breaking
> downstream clients/consumers of kubeconfig, in particular gcloud.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8344/files#r30452914>
> .
>